src/compas_timber/_fabrication/jack_cut.py

```diff
@@ -0,0 +1,276 @@
+import math
+
+from compas.geometry import BrepTrimmingError
+from compas.geometry import Frame
+from compas.geometry import Line
+from compas.geometry import Plane
+from compas.geometry import Rotation
+from compas.geometry import Vector
+from compas.geometry import angle_vectors_signed
+from compas.geometry import distance_point_point
+from compas.geometry import intersection_line_plane
+from compas.geometry import is_point_behind_plane
+
+from compas_timber.elements import FeatureApplicationError
+
+
+class BTLxProcess(object):
+    """Base class for BTLx processes.
+
+    Attributes
+    ----------
+    ref_side_index : int
+        The reference side, zero-based, index of the beam to be cut. 0-5 correspond to RS1-RS6.
+    """
+
+    def __init__(self, ref_side_index):
+        self.ref_side_index = ref_side_index
+
+
+class OrientationType(object):
+    """Enum for the orientation of the cut.
+
+    Attributes
+    ----------
+    START : int
+        The start of the beam is cut away.
+    END : int
+        The end of the beam is cut away.
+    """
+
+    START = 0
+    END = 1
+
+
+class JackRafterCut(BTLxProcess):
```

Comment:
Where does the serialization of the btlx process parameters happen? shouldn't there be a class method for that? Also does this mean that header attributes are being removed?

Reply:
You're right, that's missing. I introduced this in one of the last commits, might have not pushed yet.

I'm still very much debating though. I'd like to extract the XML schema specific stuff out of these classes, but they are nevertheless related. In the meantime, I created a `JackRafterCutParams` class whose sole responsibility is to populate the parameter values from `JackRafterCut` in they appropriate key-value pair according to the specification.

Reply:
it's in the latest commit https://github.com/gramaziokohler/compas_timber/pull/267/commits/aec27952f88eb1eb127eb97cfde185a936d68ada